File: ExtraCredit/Lab/lab1/lab1.java
import java.util.HashMap;
import java.util.Map;

public class lab1 {

    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> numMap = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];
            if (numMap.containsKey(complement)) {
                return new int[] { numMap.get(complement), i };
            }
            numMap.put(nums[i], i);
        }
        // Throw an exception if no solution is found, should not happen as per problem statement
        throw new IllegalArgumentException("No two sum solution");
    }

    public static void main(String[] args) {
        lab1 solution = new lab1();
        int[] nums = {2, 7, 11, 15};
        int target = 9;

        int[] indices = solution.twoSum(nums, target);
        System.out.println("Index1: " + indices[0] + ", Index2: " + indices[1]);
    }
}

File: ExtraCredit/Lab/lab1/lab1.py
def two_sum(nums, target):
    # Dictionary to store the difference and its index
    hashmap = {}
    # Loop through each number in the array
    for i, num in enumerate(nums):
        # Calculate the complement by subtracting the current number from target
        complement = target - num
        # If the complement is found in the hashmap, return the solution
        if complement in hashmap:
            return [hashmap[complement], i]
        # Otherwise, store the index of the current number
        hashmap[num] = i
    # Return an empty list if no solution is found, though the problem statement assures there is one
    return []

# Example usage:
nums = [2, 7, 11, 15]
target = 9
print(two_sum(nums, target))  # Output will be [0, 1] as nums[0] + nums[1] == 9
